src/main/java/io/xdb/core/state/AsyncState.java

```diff
@@ -0,0 +1,61 @@
+package io.xdb.core.state;
+
+import io.xdb.core.utils.ClassUtil;
+import io.xdb.gen.models.CommandRequest;
+import io.xdb.gen.models.StateDecision;
+import java.lang.reflect.Method;
+
+public interface AsyncState<I> {
+    /**
+     * {@link AsyncState#getInputType} is needed for deserializing data back into Java object.
+     *
+     * @return the type of the state input
+     */
+    Class<I> getInputType();
+
+    /**
+     * return a unique identifier of the state in any workflow definition.
+     *
+     * @return the state id
+     */
+    default String getId() {
```

Comment:
then similarly, we can use `AsyncStateOptions` here to merge it with our future `getAsyncStateOptions`